autorest/utility.go

```diff
@@ -138,11 +138,31 @@ func MapToValues(m map[string]interface{}) url.Values {
 	return v
 }
 
+// ToStringSlice method converts interface{} to []string.
+func ToStringSlice(s interface{}) []string {
+
+	v := reflect.ValueOf(s)
+	stringSlice := make([]string, v.Len())
+
+	for i := 0; i < v.Len(); i++ {
+		stringSlice[i] = v.Index(i).String()
+	}
+
+	return stringSlice
+}
+
 // String method converts interface v to string. If interface is a list, it
 // joins list elements using separator.
 func String(v interface{}, sep ...string) string {
 	if len(sep) > 0 {
-		return ensureValueString(strings.Join(v.([]string), sep[0]))
+		stringSlice, ok := v.([]string)
+
+		if ok == false {
+			stringSlice = ToStringSlice(v)
+		}
+		stringValue := strings.Join(stringSlice, strings.Join(sep, ""))
```

Comment:
Instead of joining all of the separators, I'd rather panic if somebody provides more than one. Honestly, this method probably needs to be deprecated and replaced soon. It is very gross.

Reply:
#211 . I would leave it with them joined until we improve the whole thing here in some future breaking change. if someone provides multiple separators the expected behavior would be to use all of them. they might see it as instead of doing `":;"` they can do `":", ";"`